test/controllers/laptops_controller_test.rb

```diff
@@ -0,0 +1,39 @@
+require 'test_helper'
+
+class LaptopsControllerTest < ActionDispatch::IntegrationTest
```

Comment:
Normally we don't do controller tests (they're very flaky). What we normally write are model and request tests/specs